sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaIO.java

```diff
@@ -1901,13 +1902,32 @@ public void processElement(OutputReceiver<KafkaSourceDescriptor> receiver) {
               for (Map.Entry<String, List<PartitionInfo>> entry :
                   consumer.listTopics().entrySet()) {
                 if (pattern.matcher(entry.getKey()).matches()) {
+                  List<PartitionInfo> partitionInfoList = entry.getValue();
+                  if (partitionInfoList == null) {
+                    LOG.warn(
+                        "Could not find any partitions info for topic {}. Please check Kafka configuration and make sure "
+                            + "that provided topics exist.",
+                        entry.getKey());
+                  }
                   for (PartitionInfo p : entry.getValue()) {
                     partitions.add(new TopicPartition(p.topic(), p.partition()));
                   }
                 }
               }
             } else {
               for (String topic : topics) {
+                try {
+                  Map<String, List<PartitionInfo>> topicsInConsumer = consumer.listTopics();
+                  checkState(
+                      topicsInConsumer.containsKey(topic),
+                      "Could not find any partitions info for topic "
+                          + topic
+                          + ". Please check Kafka configuration and make sure "
+                          + "that provided topics exist.");
+                } catch (KafkaException e) {
+                  LOG.warn(
+                      "Unable to access consumer. Skipping fail fast checks for topic {}.", topic);
+                }
```

Comment:
This is on the wrong side of the loop and will fetch all topics for every provided topic.
The return value of `partitionsFor` already returns enough information since it will be empty if the topic was not found. This will execute at runtime, so throwing exceptions must be avoided since it will only cause the element to be retried indefinitely in streaming mode.

I guess this could be replaced with:
```
final List<PartitionInfo> partitions = consumer.partitionsFor(topic);
if (partitions.isEmpty()) {
  LOG.warn("...");
  continue;
}
for (PartitionInfo p : partitions) {
```